### PR TITLE
IGAPP-948: Fixes for shelter offer

### PR DESCRIPTION
**Pull request by klinzo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

Lots of fixes in this PR. Things done:
* Fix 3m label
* Add smoking icon if "rauchen" is included in info
* Remove quarter logic as it was removed from api and form
* Add street instead of quarter to title of shelters
* Switch to new multisite api schema: https://region.wohnraum.tuerantuer.org/
* Open wohnraumbörse in inappbrowser in native apps
* Highlight Angebote in header for shelter route
* Add polish and russian translations
* Open shelters in new tabs
* Use offer from cms instead of hardcoded check
* Adjust some labels and restructure layout a little bit
* Added tooltips for icons
* Added croatian language
* Hide host type if there are zero residents

### CODE REVIEW

File: native/src/routes/Offers.tsx [hunk 1]
```diff
@@ -17,23 +17,25 @@ type PropsType = {
...
   <SpaceBetween>
     <View>
-      <Tiles title={t('offers')} tiles={toTileModels(offers)} onTilePress={navigateToOffer} language={language} />
+      <Tiles
+        title={t('offers')}
+        tiles={offers.map(offer => {
+          const isInternalExtra = [SPRUNGBRETT_OFFER_ROUTE as string].includes(offer.alias)
+          const path = isInternalExtra ? offer.alias : offer.path
+          return new TileModel({
+            title: t(offer.title),
```
Comment (klinzo) [inline native/src/routes/Offers.tsx:29]: Inlined, only change is that we try to translate the offer title.

File: api-client/src/routes/pathname.ts [hunk 2]
```diff
@@ -42,11 +42,16 @@ export const pathnameFromRouteInformation = (routeInformation: NonNullableRouteI
...
-  if (routeInformation.route === SPRUNGBRETT_OFFER_ROUTE || routeInformation.route === SHELTER_ROUTE) {
+  if (routeInformation.route === SPRUNGBRETT_OFFER_ROUTE) {
     const { cityCode, languageCode, route } = routeInformation
-    // https://integreat.app/augsburg/de/offers/sprungbrett
+    // https://integreat.app/augsburg/de/offers/wohnraum/1234, https://integreat.app/augsburg/de/offers/wohnraum
     return constructPathname([cityCode, languageCode, OFFERS_ROUTE, route])
   }
+  if (routeInformation.route === SHELTER_ROUTE) {
+    const { cityCode, languageCode, route, id } = routeInformation
+    // https://integreat.app/augsburg/de/offers/sprungbrett
```
Comment (f1sh1918) [inline api-client/src/routes/pathname.ts:52]: i think the comment should be switched with that in line 47?

File: native/src/routes/Offers.tsx [hunk 3]
```diff
@@ -2,38 +2,53 @@ import React, { ReactElement } from 'react'
...
+  t,
+  languageCode,
+  cityCode
+}: PropsType): ReactElement => (
   <SpaceBetween>
     <View>
-      <Tiles title={t('offers')} tiles={toTileModels(offers)} onTilePress={navigateToOffer} language={language} />
+      <Tiles
+        title={t('offers')}
+        tiles={offers.map(offer => {
```
Comment (f1sh1918) [inline native/src/routes/Offers.tsx:34]: can you extract this in a separate fkt. this makes the code very unreadable

File: web/src/components/CleanLink.tsx [hunk 4]
```diff
@@ -15,20 +15,29 @@ type CleanLinkProps = {
...
   className?: string
+  newTab?: boolean
 }
 
-const CleanLink: React.FC<CleanLinkProps> = ({ to, children, ariaLabel, className }: CleanLinkProps) => (
-  <>
-    {isExternalUrl(to) ? (
-      <StyledCleanLink href={to} aria-label={ariaLabel} className={className} data-testid='externalLink' as='a'>
+const CleanLink: React.FC<CleanLinkProps> = ({ to, children, ariaLabel, className, newTab }: CleanLinkProps) => {
+  const newTabProps = newTab ? { target: '_blank', rel: 'noopener noreferrer' } : {}
```
Comment (f1sh1918) [inline web/src/components/CleanLink.tsx:22]:
```suggestion
  const newTabProps = newTab && { target: '_blank', rel: 'noopener noreferrer' }
```

File: web/src/components/ShelterInformation.tsx [hunk 5]
```diff
@@ -43,7 +43,7 @@ const Detail = styled.div`
   display: flex;
 `
 
-const TooltipContainer = styled.div`
```
Comment (f1sh1918) [inline web/src/components/ShelterInformation.tsx:46]: Can you add in line 37 a box shadow for the button <3
`box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1), 0 1px 2px rgba(0, 0, 0, 0.15);`

File: web/src/components/ShelterInformationSection.tsx [hunk 6]
```diff
@@ -89,25 +90,37 @@ const ShelterInformationSection = ({
...
       </Row>
       <RowDetail viewportSmall={viewportSmall}>
-        {information.map(({ text, icon, rightText, link }) => (
-          <Detail key={text} extended={extended} as={link ? CleanLink : 'div'} to={link}>
-            {icon && <img alt='' src={icon} />}
-            <DetailText hasText={!!rightText}>{text}</DetailText>
-            {rightText && <RightTextContainer>{rightText}</RightTextContainer>}
-          </Detail>
-        ))}
+        {information.map(({ text, icon, rightText, link, tooltip }) => {
```
Comment (f1sh1918) [inline web/src/components/ShelterInformationSection.tsx:104]: its really getting hard to read here :/ But okay for the moment. Maybe extract it to separate component in the future

File: web/src/routes/OffersPage.tsx [hunk 7]
```diff
@@ -40,30 +38,24 @@ const OffersPage = ({ cityModel, cityCode, languageCode, languages }: CityRouteP
...
-      })
-
-      if (shelterOfferEnabled(cityCode)) {
-        const path = pathnameFromRouteInformation({ route: SHELTER_ROUTE, cityCode, languageCode })
-        tiles.push(new TileModel({ title: t('shelter:title'), path, thumbnail: shelterIcon }))
-      }
-
-      return tiles
-    },
+      }),
```
Comment (f1sh1918) [inline web/src/routes/OffersPage.tsx:58]: Do we show somewhere else this tile?

**Comment by f1sh1918** [comment]
I couldn't find a link to wohnraumbörse (tile before) is there no link anymore?
image
Translations in polish und russian not found
Only tested web so far. Gonna finish review tommorow

**Comment by klinzo** [comment]
> I couldn't find a link to wohnraumbörse (tile before) is there no link anymore?

Yes, its now delivered as offer from the backend. Its only activated in the testumgebung in the real cms yet and not in aux and test cms yet. Otherwise there would be two tiles showing up in the currently delivered apps with one not working.

File: web/src/routes/OffersPage.tsx (hunk 7 above)
Comment (klinzo) [reply web/src/routes/OffersPage.tsx:58]: See comment below, it is now delivered by the cms as offer.

**Review by f1sh1918 — APPROVED** [review]
tested on web and android

**Pull request merged**